src/test/features/response/routes/when-will-you-pay.ts

```diff
@@ -0,0 +1,155 @@
+          })
+
+          it('should return 500 when cannot retrieve response draft', async () => {
+            claimStoreServiceMock.resolveRetrieveClaimByExternalId()
+            draftStoreServiceMock.rejectFind('Error')
+
+            await request(app)
+              .post(defenceFullPartialPaymentOptionsPage)
+              .set('Cookie', `${cookieName}=ABC`)
+              .send(validFormData)
+              .expect(res => expect(res).to.be.serverError.withText('Error'))
+          })
+
+          it('should return 500 when cannot save response draft', async () => {
+            claimStoreServiceMock.resolveRetrieveClaimByExternalId()
+            draftStoreServiceMock.resolveFind('response')
+            draftStoreServiceMock.rejectSave()
+
+            await request(app)
+              .post(defenceFullPartialPaymentOptionsPage)
+              .set('Cookie', `${cookieName}=ABC`)
+              .send(validFormData)
+              .expect(res => expect(res).to.be.serverError.withText('Error'))
+          })
+        })
+
+        context('when service is healthy', () => {
+          beforeEach(() => {
+            claimStoreServiceMock.resolveRetrieveClaimByExternalId()
+            draftStoreServiceMock.resolveFind('response')
+          })
+
+          context('when form is valid', async () => {
+            beforeEach(() => {
+              draftStoreServiceMock.resolveSave()
+            })
+
+            async function checkThatSelectedPaymentOptionRedirectsToPage (data: object, expectedToRedirect: string) {
+              await request(app)
+                .post(defenceFullPartialPaymentOptionsPage)
+                .set('Cookie', `${cookieName}=ABC`)
+                .send(data)
+                .expect(res => expect(res).to.be.redirect.toLocation(expectedToRedirect))
+            }
+
+            it('should redirect to repayment plan page for "INSTALMENTS" option selected', async () => {
+              await checkThatSelectedPaymentOptionRedirectsToPage({ option: DefendantPaymentType.INSTALMENTS.value }, Paths.defenceFullPartialPaymentPlanPage.evaluateUri({ externalId: externalId }))
+            })
+
+            it('should redirect to pay by set date page for "FULL_BY_SPECIFIED_DATE" option selected', async () => {
+              await checkThatSelectedPaymentOptionRedirectsToPage({ option: DefendantPaymentType.FULL_BY_SPECIFIED_DATE.value }, Paths.defenceFullPartialPaymentOptionsPage.evaluateUri({ externalId: externalId }))
+            })
+          })
+
+          context('when form is invalid', async () => {
+            it('should render page', async () => {
+              await request(app)
+                .post(defenceFullPartialPaymentOptionsPage)
+                .set('Cookie', `${cookieName}=ABC`)
+                .send({ name: 'John Smith' })
+                .expect(res => expect(res).to.be.successful.withText('When will you pay?', 'div class="error-summary"'))
```

Comment:
I think it's safer to assert against text on the page. This test may unexpectedly break if e.g. someone reformats this div.

Reply:
I think text is fine

Reply:
Yeah the text is alright, I just meant to remove the `'div class="error-summary"'` bit.

Reply:
ok

Reply:
It's there to validate that there is validation errors on the page, it's a pattern everywhere in the app